animation_nodes/nodes/animation/delay_time.py

```diff
@@ -20,4 +31,31 @@ def drawLabel(self):
         else: return "Delay Time"
 
     def getExecutionCode(self, required):
-        return "outTime = time - delay"
+        if self.useListA and self.useListB:
+            return "outTimes = self.executeList_Both(times, delays)"
+        elif self.useListA and not self.useListB:
+            return "outTimes = self.executeList_Times(times, delay)"
+        elif not self.useListA and self.useListB:
+            return "outTimes = self.executeList_Delays(time, delays)"
+        else:
+            return "outTime = time - delay"
+
+    def executeList_Both(self, times, delays):
+        virtualA, virtualB = VirtualDoubleList.createMultiple((times, 0), (delays, 0))
+        amount = VirtualDoubleList.getMaxRealLength(virtualA, virtualB)
+
+        return delayTime_Multiple(virtualA, virtualB, amount)
+
+    def executeList_Times(self, times, delay):
+        delays = DoubleList.fromValue(delay)
+        virtualA, virtualB = VirtualDoubleList.createMultiple((times, 0), (delays, 0))
+        amount = VirtualDoubleList.getMaxRealLength(virtualA, virtualB)
+
+        return delayTime_Multiple(virtualA, virtualB, amount)
+
+    def executeList_Delays(self, time, delays):
+        times = DoubleList.fromValue(time)
+        virtualA, virtualB = VirtualDoubleList.createMultiple((times, 0), (delays, 0))
+        amount = VirtualDoubleList.getMaxRealLength(virtualA, virtualB)
+
+        return delayTime_Multiple(virtualA, virtualB, amount)
```

Comment:
All of those methods are identical. You can just execute one of them if if any of the inputs is a list. Also, avoid using the names like A and B. We know exactly what those lists contain and they should be named accordingly.